src/json-patch.ts

```diff
@@ -136,30 +189,29 @@ module jsonpatch {
       obj[key] = this.value;
       return removed;
     },
-    move: function (obj, key, tree) {
-      var getOriginalDestination: any = { op: "_get", path: this.path };
-      apply(tree, [getOriginalDestination]);
+    move: function (obj, key, document) {
+      const originalValue = jsonpatch.getValueByPointer(document, this.path);
       // In case value is moved up and overwrites its ancestor
-      var original = getOriginalDestination.value === undefined ?
-        undefined : JSON.parse(JSON.stringify(getOriginalDestination.value));
+      var original = originalValue === undefined ?
+        undefined : JSON.parse(JSON.stringify(originalValue));
```

Comment:
This was necessary when we replaced `root` in place by iterating all the keys and replacing them with `op.value` keys. This way original value would be altered and this breaks our Polymer notification. 

But now we don't do that so I removed cloning.